compiler/native/compile.go

```diff
@@ -5,14 +5,86 @@
 package native
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"net/http"
 	"strings"
 
+	"github.com/go-vela/types/library"
 	"github.com/go-vela/types/pipeline"
+	"github.com/go-vela/types/yaml"
 )
 
+// ModifyRequest contains the payload passed to the modification endpoint
+type ModifyRequest struct {
+	Pipeline *yaml.Build `json:"pipeline,omitempty"`
+	Build    int         `json:"build,omitempty"`
+	Repo     string      `json:"repo,omitempty"`
+	User     string      `json:"user,omitempty"`
+}
+
+// modifyConfig sends the configuration to external http endpoint for modification.
+func (c *client) modifyConfig(build *yaml.Build, libraryBuild *library.Build, repo *library.Repo) (*yaml.Build, error) {
```

Comment:
A bit nitpicky, but I think in general we put private functions at the bottom of Go files.